docs/guides/elements-lookup-troubleshooting.md

```diff
@@ -0,0 +1,123 @@
+whose nesting level is deeper than `62`. This limitation has to do with how `NSDictionary` works and cannot be worked
+around. The default maximum nesting level for the XCUITest driver is set to `50` and could be customized by the
+[snapshotMaxDepth](../reference/settings.md) setting.
+[React Native](https://reactnative.dev/) is known to create
+such deep hierarchies and the only viable solution for now is to fix the application
+under test by flattening nested views. Check the corresponding [issue](https://github.com/appium/appium/issues/14825)
+for more details.
+Deeply nested hierarchies might also be the reason for the element lookup slowness. Read the [Diagnosing WebDriverAgent Slowness](./wda-slowness.md) article to troubleshoot the latter.
+
+
+## Symptom #2
+
+The desired element is shown in the page tree, but cannot be found if looked up from an automated test.
+
+## Resolutions To Symptom #2
+
+### Make sure there is no race condition
+
+Sometimes the automation might too fast or too slow depending on in which state the UI is while the lookup is being
+executed. If it is too fast then consider using lookup timers, e.g. repeat the `findElement` more than once until
+either the element is found or the timeout occurs. All clients have convenience wrappers for such timers in form of
+expected conditions.
+If the automation is too slow, e.g. the desired element disappears faster than `findElement` could detect its presence
+then make sure your script is optimized for the maximum performance, e.g. optimal/fast element locators are used,
+the application itself and driver settings are [adjusted](./wda-slowness.md) to perform optimally, etc.
+There might be situations where the automation framework is already optimized, although the desired element is
+a short-living one, for example some notification popup that only appears for a second and then is immidiely hidden.
+For such "special" elements consider using approaches different from `findElement`, for example post-test video recording analysis (video FPS should usually be enough to catch all short-living elements), or introducing special
+application debug settings to change the behavior for such elements and make them stay visible for longer time, or
+using non-UI-related assertions, like logs analysis or direct API calls.
+
+### Make sure the debug environment matches to the testing one
+
+There are known cases where application interface/behavior might differ in simulators and real devices. It might even differ
+if the screen size or device model/OS version/system setting differs. That is why always make sure your debug
+environment, for example one where Appium Inspector is used,
+is as close as possible to the environment where automated tests are being executed.
+
+
+## Symptom #3
+
+The desired element is shown in the page tree, but its property value is not what expect it to be, for example it
```

Comment:
Can be simplified to '[..] value is not as expected, [..]'

Reply:
changed